src/codecs/format/bytes.rs

```diff
@@ -31,19 +31,23 @@ impl DeserializerConfig for BytesDeserializerConfig {
 /// This deserializer can be considered as the no-op action for input where no
 /// further decoding has been specified.
 #[derive(Debug, Clone)]
-pub struct BytesDeserializer;
+pub struct BytesDeserializer {
+    log_schema_message_key: &'static str,
+}
 
 impl BytesDeserializer {
     /// Creates a new `BytesDeserializer`.
-    pub const fn new() -> Self {
-        Self
+    pub fn new() -> Self {
+        Self {
+            log_schema_message_key: log_schema().message_key(),
+        }
     }
 }
 
 impl Deserializer for BytesDeserializer {
     fn parse(&self, bytes: Bytes) -> crate::Result<SmallVec<[Event; 1]>> {
         let mut log = LogEvent::default();
-        log.insert(log_schema().message_key(), bytes);
+        log.insert_flat(self.log_schema_message_key, bytes);
```

Comment:
We can decide to introduce this restriction, but there is a behavior change here. If a user configured their global `message_key` to be something like `foo.bar`, it would have previously created inserted a map at `foo` with a `bar` key with the value, now it will just insert a single `foo.bar` key.

Reply:
My question would be: how much does this give us? If not much, I'd say leave it out and maintain behavior, if significant, I'd say we can think about restricting the global schema feature to flat keys.

Reply:
Ah, hmm. This function -- `parse` -- represents the serialization point for the `codec_perf` program you can find in https://github.com/vectordotdev/vector/pull/10581/commits/19f292cbd4b108df672294a6934b2ac2dd7e6ea1 and the serialization point here is `insert`. Going through path iteration code is just slow, slow, slow. Once the soaks run, if there's an improvement, I can flip this back to `insert` and see what the change in numbers are.

Reply:
We also know, at Vector start-time, if the specified log schema key has a `.` in it so this also seems like a problem we could solve by either:

* encoding the information about whether the log schema key is flat or not to use to decide how to insert
* add a "fast-path" to `#insert` that just scans for a `.` to decide what to do. This is suboptimal since we'd do it every time we insert, even though we know at Vector start time if the key is flat or not, but is a smaller change and the optimization may still be faster (though not as fast as a straight `insert_flat`)

Reply:
Considering this change didn't move the soak numbers I've flipped this back to `insert`.